sol/main.c

```diff
@@ -664,20 +664,22 @@ run_sol(struct net_config *net_conf, struct sol_config *sol_conf)
  * There should be only one sol_config instance.
  * Return an error if trying to allocate the second instance.
  *
- * Use rte_calloc() to zero-out the instance and initialize the
+ * Use rte_calloc_socket() to zero-out the instance and initialize the
  * request queue list to guarantee that cleanup_sol() won't fail
  * during initialization.
  */
 struct sol_config *
-alloc_sol_conf(void)
+alloc_sol_conf(unsigned int lcore)
 {
 	struct sol_config *sol_conf;
 	static rte_atomic16_t num_sol_conf_alloc = RTE_ATOMIC16_INIT(0);
 	if (rte_atomic16_test_and_set(&num_sol_conf_alloc) > 1) {
```

Comment:
According to the DPDK documentation, the return value of [rte_atomic16_test_and_set](https://doc.dpdk.org/api/rte__atomic_8h.html#aac272d019db176feebe240dc14f1efe4) is either 0 or 1: ``If the counter value is already set, return 0 (failed). Otherwise, set the counter value to 1 and return 1 (success).`` So the condition check here is always invalid.